bootstrap/kubeadm/api/v1alpha3/kubeadmconfig_types.go

```diff
@@ -19,7 +19,7 @@ package v1alpha3
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
-	kubeadmv1beta1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/types/v1beta1"
+	kubeadmv1beta2 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/types/v1beta2"
```

Comment:
@gab-satchi 
I feel really bad about this because I opened the issue behind this PR, but now that I see that removing v1beta1 is impacting v1alpha3 I don't think we can move on with this change, otherwise we can't ensure a clean v1alpha3 --> v1alpha4 upgrade.
This change has to wait for v1alpha3 being removed before 😞 
Sorry again for not having considered this when opening the issue
/hold

Reply:
We should probably mostly drop support / deprecate in this release, rather than removing the APIs altogether. In the next release, we should be able to proceed with the complete removal, alongside the v1alpha3 API types.

Reply:
No worries @fabriziopandini. let me know if it makes more sense to close and re-open when the time is right